kraken/pod_scenarios/setup.py

```diff
@@ -34,3 +37,89 @@ def run(kubeconfig_path, scenarios_list, config, failed_post_scenarios, wait_dur
+                else:
+                    pre_action_output = ""
+                container_killing_in_pod(cont_scenario)
+                logging.info("Waiting for the specified duration: %s" % (wait_duration))
+                time.sleep(wait_duration)
+                failed_post_scenarios = post_actions.check_recovery(
+                    kubeconfig_path, container_scenario_config, failed_post_scenarios, pre_action_output
+                )
+                cerberus.publish_kraken_status(config, failed_post_scenarios)
+                logging.info("")
+
+
+def container_killing_in_pod(cont_scenario):
+
+    namespace = cont_scenario.get("namespace", "*")
+    label_selector = cont_scenario.get("label_selector", None)
+    pod_names = cont_scenario.get("pod_names", [])
+    container_name = cont_scenario.get("container_name", "")
+
+    if type(pod_names) != list:
+        logging.error("Please make sure your pod_names are in a list format")
+        sys.exit(1)
+    if len(pod_names) == 0:
+        if namespace == "*":
+            # returns double array of pod name and namespace
+            pods = kubecli.get_all_pods(label_selector)
+        else:
+            # Only returns pod names
+            pods = kubecli.list_pods(namespace, label_selector)
+    else:
+        if namespace == "*":
+            logging.error("You must specify the namespace to kill a container in a specific pod")
+            sys.exit(1)
+        pods = pod_names
+    logging.info("pods names " + str(pods))
+    # get container and pod name
+    container_pod_list = []
+    for pod in pods:
+        if type(pod) == list:
+
+            container_names = runcommand.invoke(
+                'oc get pods %s -n %s -o jsonpath="{.spec.containers[*].name}"' % (pod[0], pod[1])
+            ).split(" ")
+            container_pod_list.append([pod[0], pod[1], container_names])
+        else:
+            container_names = runcommand.invoke(
+                'oc get pods %s -n %s -o jsonpath="{.spec.containers[*].name}"' % (pod, namespace)
+            ).split(" ")
+            container_pod_list.append([pod, namespace, container_names])
+
+    # jsonpath="{.items[*].spec.containers[*].name}"
+    for container_pod in container_pod_list:
+        for c_name in container_pod[2]:
+            if container_name != "":
+                logging.info("c name " + str(c_name) + " " + str(container_name))
+                if c_name == container_name:
+                    retry_container_killing(container_pod[0], container_pod[1], c_name)
+            else:
+                retry_container_killing(container_pod[0], container_pod[1], c_name)
+
+
+def retry_container_killing(podname, namespace, container_name):
+    i = 0
+    while i < 5:
+        logging.info("Killing container %s in pod %s (ns %s)" % (str(container_name), str(podname), str(namespace)))
+        response = kubecli.exec_cmd_in_pod("kill 1", podname, namespace, container_name)
```

Comment:
Maybe expose the kill signal as part of the config? Thoughts?

Reply:
+1 for this.   Many OCP pods protect themselves by catching SIGINT.    Default to -2/SIGINT but allow arbitrary signals.   Or even random signals for more chaos fun!

Reply:
Sounds good! Will add this to the configs

Reply:
PTAL at my updates